src/3d/qgsrubberband3d.cpp

```diff
@@ -140,6 +140,32 @@ void QgsRubberBand3D::setupPolygon( Qt3DCore::QEntity *parentEntity )
   mPolygonEntity->addComponent( mPolygonMaterial );
 }
 
+void QgsRubberBand3D::removePoint( int index )
+{
+  if ( QgsPolygon *polygon = qgsgeometry_cast<QgsPolygon *>( mGeometry.constGet() ) )
```

Comment:
You're abusing API here. You're using constGet(), but then modifying the geometry! :scream: 

Bit this points out a bad issue -- you shouldn't be allowed to qgsgeometry_cast a const geometry pointer to a non-const one. PR incoming...